openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -266,7 +267,9 @@ def set_credit_requirement_status(username, course_key, req_namespace, req_name,
     # If we're marking this requirement as "satisfied", there's a chance
     # that the user has met all eligibility requirements.
     if status == "satisfied":
-        CreditEligibility.update_eligibility(reqs, username, course_key)
+        is_eligible, eligibility_record_created = CreditEligibility.update_eligibility(reqs, username, course_key)
```

Comment:
Could you just return `is_eligible` here and check that?  I'm not sure why you need both values.